fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/ExpressionRef.java

```diff
@@ -32,12 +32,13 @@
 
 /**
  * This interface is used mostly as an (admittedly surmountable) barrier to rules mutating bound references directly,
- * which is undefined behavior. Unlike {@link MutableExpressionRef}, it does not provide an <code>insert()</code> method,
- * so the rule would need to cast the reference back to {@link MutableExpressionRef} before modifying it.
+ * which is undefined behavior.
  * @param <T> the type of planner expression that is contained in this reference
  */
 @API(API.Status.EXPERIMENTAL)
 public interface ExpressionRef<T extends PlannerExpression> extends Bindable {
```

Comment:
Yes, that's right. There's a reason, but it might not be a good one.

The reason to keep it is in the Javadoc:
> This interface is used mostly as an (admittedly surmountable) barrier to rules mutating bound references directly.

The idea is that it provides some kind of discouragement to a rule writer who tries to invoke the various methods in `GroupExpressionRef`, which will totally confuse the planner. I'm not sure if the overhead is worth it, though.